src/factotum_server/mod.rs

```diff
@@ -0,0 +1,242 @@
+use std::thread::JoinHandle;
+use iron::prelude::*;
+use iron::typemap::Key;
+use logger::Logger;
+use persistent::{Read, State};
+use threadpool::ThreadPool;
+
+use Args;
+use factotum_server::command::{CommandStore, Execution};
+use factotum_server::dispatcher::{Dispatch, Dispatcher, Query};
+use factotum_server::persistence::{Persistence, ConsulPersistence, JobState};
+use factotum_server::responder::{DispatcherStatus, JobStatus, WorkerStatus};
+use factotum_server::server::{ServerManager, JobRequest};
+
+#[derive(Debug, Copy, Clone)]
+pub struct Server;
+impl Key for Server {
+    type Value = ServerManager;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct Storage;
+impl Key for Storage {
+    type Value = ConsulPersistence;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct Paths;
+impl Key for Paths {
+    type Value = RwLock<CommandStore>;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct Updates;
+impl Key for Updates {
+    type Value = Mutex<Sender<Dispatch>>;
+}
+
+pub fn start(args: Args) {
+    let server = ServerManager::new(args.flag_ip, args.flag_port, args.flag_webhook, args.flag_no_colour);
+    let persistence = ConsulPersistence::new(args.flag_consul_name, args.flag_consul_ip, args.flag_consul_port, args.flag_consul_namespace);
+    let dispatcher = Dispatcher::new(args.flag_max_jobs, args.flag_max_workers);
+    let command_store = commands![::FACTOTUM.to_string() => args.flag_factotum_bin];
+    
+    let address = SocketAddr::from_str(&format!("{}:{}", server.ip, server.port)).expect("Failed to parse socket address");
+
+    let (requests_channel, _, _) = trigger_worker_manager(dispatcher, persistence.clone(), &command_store).unwrap();
+
+    let router = router!(
+        index:      get     "/"         =>  responder::api,
+        help:       get     "/help"     =>  responder::api,
+        status:     get     "/status"   =>  responder::status,
+        settings:   post    "/settings" =>  responder::settings,
+        submit:     post    "/submit"   =>  responder::submit,
+        check:      post    "/check"    =>  responder::check
+    );
+    let (logger_before, logger_after) = Logger::new(None);
+
+    let mut chain = Chain::new(router);
+    chain.link_before(logger_before);
+    chain.link(State::<Server>::both(server));
+    chain.link(State::<Storage>::both(persistence));
+    chain.link(Read::<Paths>::both(RwLock::new(command_store)));
+    chain.link(Read::<Updates>::both(Mutex::new(requests_channel)));
+    chain.link_after(logger_after);
+    
+    match Iron::new(chain).http(address) {
+        Ok(listening) => {
+            let socket_addr = listening.socket;
+            let ip = socket_addr.ip();
+            let port = socket_addr.port();
+            let start_message = format!("Factotum Server version [{}] listening on [{}:{}]", ::VERSION, ip, port);
+            info!("{}", start_message);
+            println!("{}", start_message)
+        }
+        Err(e) => println!("Failed to start server - {}", e)
```

Comment:
Yep ok - see what you mean. My point was only that `start(...)` may or may not start something if it has an error.